### PR TITLE
Dev #17242: Remove unused invocations to getOwnQuestionAttributesViaPlugin() and some QuestionBaseDataSet methods

### PR DESCRIPTION
**Pull request by gabrieljenik**
Remove getPreformattedBlockOfAdvancedSettings and getAdvancedOptions from QuestionBaseDataSet

Move getAttributesAsArrayFromDB to its own method 
Move \QuestionTheme::getAdditionalAttrFromExtendedTheme to its own method 
Move \QuestionAttribute::getQuestionAttributesSettings to its own method

Removed Hungarian notation

### CODE REVIEW

File: application/models/services/QuestionAttributeHelper.php [hunk 1]
```diff
@@ -89,176 +90,213 @@ public function sanitizeQuestionAttributes($aAttributes)
...
+
+    /**
+     * Get all saved attribute values for one question as array
+     *
+     * @param int $questionId  the question id
+     * @return array the returning array structure will be like
+     *               array(attributeName => array(languageCode => value, ...), ...)
+     *               where languageCode is '' if no language is specified.
+     */
+    private function getAttributesValuesFromDB($questionId)
```
Comment (olleharstedt) [inline application/models/services/QuestionAttributeHelper.php:274]: Fun fact: Cannot be private, because private methods cannot be mocked. :(

**Review by olleharstedt — CHANGES_REQUESTED** [review]

**Comment by Shnoulle** [comment]
> Removed Hungarian notation

Not related to the commit : when we decide to not use Hungarian notation ? 
I dislike to use it, but during a time : it's asked :)

**Comment by olleharstedt** [comment]
> Not related to the commit : when we decide to not use Hungarian notation ?
> I dislike to use it, but during a time : it's asked :)

It's been discussed lightly a couple of times. I finally had enough, and decided to ban it in new service class code. ><

Compare with variable and function naming section in quality guide.

**Comment by Shnoulle** [comment]
Must update : https://manual.limesurvey.org/Coding_guidelines#Naming_convention

Quality guide are not in https://manual.limesurvey.org/Development_overview

**Comment by Shnoulle** [comment]
> It's been discussed lightly a couple of times. 

I didn't remind : dev meeting ?

**Comment by olleharstedt** [comment]
> I didn't remind : dev meeting ?

Internally, sorry.

**Comment by olleharstedt** [comment]
> Must update : https://manual.limesurvey.org/Coding_guidelines#Naming_convention
> 
> Quality guide are not in https://manual.limesurvey.org/Development_overview

Yes, since my new guide is still in draft, I didn't remove it. But maybe remove the Hungarian notation at least.

**Comment by Shnoulle** [comment]
> 
> Yes, since my new guide is still in draft, I didn't remove it. But maybe remove the Hungarian notation at least.

Maybe a link with information about reworking on this page and the new page (in DRAFT) ?

**Comment by Shnoulle** [comment]
Old eyes must not be made fun of please

**Comment by olleharstedt** [comment]
> Old eyes must not be made fun of please

:smile:

**Pull request merged**